### ISSUE TITLE
nix derivation

### ISSUE THREAD

**Issue by Kloenk** [o2sh/onefetch#109]
I would like a Nix derivation, so one can build with nix-build, and maybe someone could even get it into nixpkgs, so it is available on any system running nix.
This feature would need a Cargo.lock file, and a default.nix, I will see that I build that files

**Comment by o2sh** [comment]
@Kloenk the NixOs [build ](https://travis-ci.org/o2sh/onefetch) just failed, any idea ?

**Comment by Kloenk** [comment]
yep, new rust source commits, but the sha256 sum in the default.nix was not updated.

line 19 in default.nix must have the value ```0cpfjxn6nqsrnmgvjzr17n6xbbyfl9c91d5kbrmxb6jyig98k6aq```

**Comment by spenserblack** [comment]
@Kloenk, does this mean `default.nix` will need to be updated each time the master branch has been updated?

**Comment by Kloenk** [comment]
yes, because of the vendor sha256 sum

**Comment by o2sh** [comment]
yeah, we cannot keep that on the master branch.
Maybe create a second branch dedicated to test NixOs build.

But I think we should rollback the changes on the travis.yml file

**Comment by spenserblack** [comment]
Hmmm... 🤔 Perhaps it's best to *not* test the nix-build in the Travis build. At least not on the master branch. If nix-build must be pointed to specific commits, it would be irrelevant for testing PRs (since nix-build would be building a SHA, not the latest commit in the PR), and the build would fail *at least* once after each merged PR, since I don't think we can expect the maker of the PR to fix `default.nix`.

A possible solution I can think of is to have a separate branch just for nix-build. Set Travis to *only* test nix-build on this branch.

**Comment by o2sh** [comment]
@spenserblack we're in sync 😄

**Comment by Kloenk** [comment]
I agree that there is no need for the ```nix-build``` test. 

If we delete the default.nix that's also fine with me, but please keep the Cargo.lock file updated, at least with every tagged version, because that is needed for a release in nixpkgs. the default.nix is not needed in this repo for the package there.

**Comment by spenserblack** [comment]
Makes sense to me 👍 
I agree that `Cargo.lock` should remain tracked. I believe it's standard for binaries to have their lock files tracked when available.

**Comment by Kloenk** [comment]
I created a pr on nixpkgs for version 1.7.0

nixos/nixpkgs#72254

**Comment by filalex77** [comment]
@Kloenk I very much appreciate the effort you did in your PR, but there were a couple issues I found:
- your `default.nix` uses some unusual idioms, indicated by the reviewers' comments
- it specifies version 1.7.0, whether the current latest is 2.1.0
- having `default.nix` in `onefetch`'s repository doesn't help a lot - `nixpkgs` uses `rustPlatform` uniformly
- there are some conflicts with the `master` branch as indicated by GitHub

I decided to create a new PR incorporating all these changes: https://github.com/NixOS/nixpkgs/pull/74191

I consider it important for you to get all the necessary attribution. So, I added you as a co-author to my PR - that should count. Feel free to add yourself as a maintainer if you feel like it (probably in a separate PR, as you need to add yourself to the maintainers list beforehand).

If you have any questions - reach out to me. Thanks for your contributions!
